hswidget/cofwidget/widget.go

```diff
@@ -71,6 +71,51 @@ func (p *widget) Build() {
 	}
 }
 
+func (p *widget) getStateID() string {
+	return fmt.Sprintf("widget_%s", p.id)
+}
+
+func (p *widget) getState() *widgetState {
+	var state *widgetState
+
+	s := giu.Context.GetState(p.getStateID())
+
+	if s != nil {
+		state = s.(*widgetState)
+		if len(p.cof.CofLayers) > 0 {
+			state.viewerState.layer = &p.cof.CofLayers[state.viewerState.layerIndex]
+		}
+	} else {
+		p.initState()
+		state = p.getState()
+	}
+
+	return state
+}
+
+func (p *widget) setState(s giu.Disposable) {
+	giu.Context.SetState(p.getStateID(), s)
+}
+
+func (p *widget) initState() {
+	state := &widgetState{
+		mode: modeViewer,
+		viewerState: &viewerState{
+			confirmDialog: &hswidget.PopUpConfirmDialog{},
+		},
+		newLayerFields: &newLayerFields{
+			selectable: true,
+			drawEffect: int32(d2enum.DrawEffectNone),
+		},
+	}
+
+	if len(p.cof.CofLayers) > 0 {
+		state.viewerState.layer = &p.cof.CofLayers[0]
+	}
```

Comment:
please move this into `hswidget/cofwidget/state.go

Reply:
There is no need of doing this. You have already moved it.

Reply:
@1Dz87  but your change are still necessary in state.go

Reply:
@gucio321 It's there... You moved it.
`	if len(p.cof.CofLayers) > 0 {
		state.viewerState.layer = &p.cof.CofLayers[0]
	}`

Reply:
@1Dz87 I meant, that I moved the getState method into state.go, but the following code, you added here, is still necessary in state.go (below this line: https://github.com/OpenDiablo2/HellSpawner/blob/5f782226b49ea120a74f909dfc2b224db31dc843/hswidget/cofwidget/state.go#L199):
```go
         if len(p.cof.CofLayers) > 0 {
		state.viewerState.layer = &p.cof.CofLayers[0]
	}
```

Reply:
Oh, sorry! A bit misunderstood( Now done.